torch/csrc/jit/mobile/code.h

```diff
@@ -27,7 +27,7 @@ struct Code {
   // We may need a two-stage importing scheme, where we firstly construct all
   // function objects, and then append referenced function pointers. This could
   // be done in parseMethods().
-  std::vector<mobile::Function*> functions_;
+  std::vector<std::unique_ptr<mobile::Function>> functions_;
```

Comment:
I see your point. For your case I think it's feasible to let `Code` hold upgrader ownerships. However there are two downsides:
1. If code holds unique_ptr to other functions, we cannot support recursive calls in the future.
2. More importantly, if multiple functions use the same upgrader functions, we need to create multiple instances of the same upgrader function.

Therefore when I wrote `Code` I reached the conclusion that `Code` doesn't own `Function`s. It's okay to use raw pointer to express the non-ownership even in modern C++.

The alternative solution is finding a place to hold these functions and just borrow to `Code`s. Something like:
```
Function& Function::get() {
    static std::vector<std::unique_ptr<Function>> funcHolder;
    
    funcHolder.emplace_back(...);
    auto& func = funcHolder.back();
    
    return func;
}
```

Reply:
Address in the latest comment